packages/react-native/scripts/cocoapods/utils.rb

```diff
@@ -196,7 +197,14 @@ def self.apply_xcode_15_patch(installer, xcodebuild_manager: Xcodebuild)
             end
             project.save()
         end
+    end
+
+    def self.apply_spm_dependencies(installer)
+        SPM.apply(installer)
+    end
 
+    def self.spm_dependency(s, url:, requirement:,  products:)
+        SPM.dependency(s, url: url, requirement: requirement,  products: products)
     end
```

Comment:
To simplify the usage for 3rd party dependencies, I'd properly move this function from the `utils.rb` directly to the `react_native_pods`. 
In this way, instead of the check:

```
if const_defined?(:ReactNativePodsUtils) && ReactNativePodsUtils.respond_to?(:spm_dependency) 
  ReactNativePodsUtils.spm_dependency(s,  
    url: 'https://github.com/apple/swift-atomics.git', 
    requirement: {kind: 'upToNextMajorVersion', minimumVersion: '1.1.0'}, 
    products: ['Atomics'] 
  ) 
else 
  raise "Please upgrade React Native to >=0.75.0 to use SPM dependencies." 
end 
```

they can just:
```
if defined?(:spm_dependency)
  spm_dependency(s,  
    url: 'https://github.com/apple/swift-atomics.git', 
    requirement: {kind: 'upToNextMajorVersion', minimumVersion: '1.1.0'}, 
    products: ['Atomics'] 
  )
else 
  raise "Please upgrade React Native to >=0.75.0 to use SPM dependencies." 
end 
```

I would also consider backporting this to all the RN versions in the supported window (right now: 0.72, 0.73, 0.74) so that they don't even need to `if-else`.

What do you think?

Reply:
Added as `spm_dependency`. 

As noted https://github.com/facebook/react-native/pull/44627#issuecomment-2123119711 this approach does depend on what linking (static/dynamic) the RN framework uses and what linking does the SPM product uses. This is something to consider with backporting, as it might require some further changes to be usefull in all scenarios